### PR TITLE
Admin Ui Alpha v1

### PR DESCRIPTION
**Pull request by gallayl**
## Done
 - [x] Update layout
   - [x] Add setting to swipeable temporary / modal drawer, update mobile defaults
   - [x] Move avatar / user info / personal settings editor to drawer
 - [x] Create / Upload content
   - [x] Add FAB
   - [x] Drawer for selecting type + Upload
   - [x] Upload on file drop
   - [x] New content modal
 - [x] Change repository
   - [x] Individual session management
   - [x] Repository switcher component
 - [x] "More" actions
   - [x] Content info
   - [x] Delete
   - [x] Edit properties
---
## Next steps
 - [ ] Localization
 - [ ] Extract default Document Viewer settings
 - [ ] fixes is @sensenet/controls-react
 - [ ] monaco on mobile - alternative? fallback?
 - [ ] EditContent - Move from Redux to hooks
 - [ ] Clean up AuthorizedRoute(s)
 - [ ] Breadcrumbs component to generic
 - [ ] Extract Props from to interfaces
 - [ ] Update AllowedChildTypes in Add view
 - [ ] CommandPalette improvements
   - [ ] Check if can be moved from Redux to hooks
   - [ ] Provider for context menu actions on current content
   - [ ] Provider for navigation entries (Go to e.g. Personal Settings, Dashboard, etc...)
 - [ ] Wire Content Picker
    - [ ] Move content (from context menu)
    - [ ] Copy content (from context menu)
 - [ ] Dirty form checking (on edit properties + edit binary)
 - [ ] Search & saved queries
 - [ ] Version info
 - [ ] Setup (settings, needs UX)
 - [ ] Dashboard
 - [ ] Upload progress info
 - [ ] Info about failed requests / Event log
 - [ ] Improve error boundaries - feedback tool?
 - [ ] Extract ContentListPanel keyboard handler - move to ContentList component(?)

### CODE REVIEW

File: apps/sensenet/src/services/MonacoModels/PersonalSettingsModel.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                type: 'string',
+                description: 'The path for the repository, e.g.: https://my-sensenet-repository.org',
+              },
+              loginName: {
+                type: 'string',
+                description: "The last user you've logged in with",
+              },
+              displayName: {
+                type: 'string',
+                description: 'An optional user friendly name to display when referencing this repository',
```
Comment (herflis) [inline apps/sensenet/src/services/MonacoModels/PersonalSettingsModel.ts]:
```suggestion
                description: 'An optional user friendly name to display the repository when you're connected to it',
```

File: apps/sensenet/src/context/PersonalSettingsContext.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { useContext, useEffect, useState } from 'react'
```
Comment (B3zo0) [inline apps/sensenet/src/context/PersonalSettingsContext.tsx]:
```suggestion
import React. { useContext, useEffect, useState } from 'react'
```

File: apps/sensenet/src/context/PersonalSettingsContext.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import React from 'react'
```
Comment (B3zo0) [inline apps/sensenet/src/context/PersonalSettingsContext.tsx]:
```suggestion
```

File: apps/sensenet/src/context/SessionContext.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
+import { ConstantContent, LoginState } from '@sensenet/client-core'
+import { debounce } from '@sensenet/client-utils'
+import { Group, User } from '@sensenet/default-content-types'
+import { useContext, useEffect, useState } from 'react'
```
Comment (B3zo0) [inline apps/sensenet/src/context/SessionContext.tsx]:
```suggestion
import React, { useContext, useEffect, useState } from 'react'
```

File: apps/sensenet/src/context/SessionContext.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import React from 'react'
```
Comment (B3zo0) [inline apps/sensenet/src/context/SessionContext.tsx]:
```suggestion
```

File: apps/sensenet/src/store/ActionHelpers.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+export const createAction = <TReturns extends { type: string }, TArgs extends any[]>(
```
Comment (B3zo0) [inline apps/sensenet/src/store/ActionHelpers.ts]: This file is not needed anymore. You can import these helpers from sn-redux.

File: apps/sensenet/src/utils/deepMerge.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import { DeepPartial } from 'redux'
```
Comment (B3zo0) [inline apps/sensenet/src/utils/deepMerge.ts]: Would it make sense to move this to sn-client-utils? Don't we import a lib that has a function like this?

File: apps/sensenet/index.html [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+        background: #888;
+      }
+
+      /* Handle on hover */
+      ::-webkit-scrollbar-thumb:hover {
+        background: #555;
+      }
+    </style>
+    <link href="https://cdnjs.cloudflare.com/ajax/libs/normalize/8.0.1/normalize.min.css" rel="stylesheet" />
+    <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500" />
```
[the remark this answers is not in the document]
Comment (gallayl) [reply apps/sensenet/index.html]: Roboto is required by [@material-ui](https://material-ui.com/). The other two will be removed.

File: apps/sensenet/package.json [hunk 9]
```diff
@@ -0,0 +1,110 @@
+{
+  "name": "@sensenet/sn-app",
+  "version": "1.3.1",
```
[the remark this answers is not in the document]
Comment (gallayl) [reply apps/sensenet/package.json]: 👍

File: apps/sensenet/package.json [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+  "main": "dist/index.js",
+  "files": [
+    "dist",
+    "src"
+  ],
+  "keywords": [
+    "sensenet",
+    "ecm",
+    "react",
+    "content list"
```
[the remark this answers is not in the document]
Comment (gallayl) [reply apps/sensenet/package.json]: 👍

File: apps/sensenet/src/components/drawer/TemporaryDrawer.tsx [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+import { UserAvatar } from '../UserAvatar'
+import { getAllowedDrawerItems } from './Items'
+
+const TemporaryDrawer: React.StatelessComponent<RouteComponentProps & { isOpened: boolean }> = props => {
+  const settings = useContext(ResponsivePersonalSetttings)
+  const personalSettings = useContext(PersonalSettingsContext)
+  const repo = useContext(RepositoryContext)
+  const theme = useContext(ThemeContext)
+  const session = useContext(SessionContext)
+  const [opened, setOpened] = useState(false)
```
[the remark this answers is not in the document]
Comment (gallayl) [reply apps/sensenet/src/components/drawer/TemporaryDrawer.tsx]: THX for pointing out, this, I've refactored the opening / closing logic to use isOpened from props, also added an onClise and onOpen callback

File: apps/sensenet/src/store/ActionHelpers.ts (hunk 6 above)
Comment (gallayl) [reply apps/sensenet/src/store/ActionHelpers.ts]: 👍

File: apps/sensenet/tsconfig.test.json [hunk 12]
```diff
@@ -0,0 +1,8 @@
+{
```
[the remark this answers is not in the document]
Comment (gallayl) [reply apps/sensenet/tsconfig.test.json]: 🆗

File: apps/sensenet/src/utils/deepMerge.ts (hunk 7 above)
Comment (gallayl) [reply apps/sensenet/src/utils/deepMerge.ts]: 👍

**Pull request merged**